src/main/java/io/dockstore/provision/GSPlugin.java

```diff
@@ -0,0 +1,300 @@
+            String blobName = getBlobName(sourcePath);
+
+            Blob blobMetadata = gsClient.get(bucketName, blobName, Storage.BlobGetOption.fields(Storage.BlobField.values()));
+            long inputSize = blobMetadata.getSize();
+
+            Blob blob;
+            try {
+                blob = gsClient.get(BlobId.of(bucketName, blobName));
+            } catch (StorageException e) {
+                System.err.println("gsClient get download exception:" + e.getMessage());
+                return false;
+            }
+
+            // The destination path does not exist when a WDL workflow
+            // or tool is run so make sure the directories in the path exist
+            try {
+                Files.createDirectories(destination.getParent());
+            } catch (IOException e) {
+                System.err.println("Could not create destination path. IO exception:" + e.getMessage());
+                return false;
+            }
+
+            ProgressPrinter printer = new ProgressPrinter();
+
+            FileOutputStream fileContent;
+            try {
+                fileContent = new FileOutputStream((destination.toFile()));
+            } catch (FileNotFoundException e) {
+                System.err.println("File not found exception:" + e.getMessage());
+                return false;
+            }
+
+            PrintStream writeTo = new PrintStream(fileContent);
+
+            if (blob.getSize() < MAX_FILE_SIZE_FOR_SINGLE_WRITE) {
+                // Blob is small read all its content in one request
+                byte[] content = blob.getContent();
+                try {
+                    writeTo.write(content);
+                } catch (IOException e) {
+                    System.err.println("Could not write download file. IO exception:" + e.getMessage());
+                    return false;
+                }
+            } else {
+                // When Blob size is big or unknown use the blob's channel reader.
+                try (ReadChannel reader = blob.reader()) {
+                    WritableByteChannel channel = Channels.newChannel(writeTo);
+                    ByteBuffer bytes = ByteBuffer.allocate(DOWNLOAD_BUFFER_SIZE);
+                    try {
+                        int limit;
+                        long runningTotal = 0;
+                        while ((limit = reader.read(bytes)) > 0) {
+                            bytes.flip();
+                            channel.write(bytes);
+                            bytes.clear();
+                            runningTotal = runningTotal + limit;
+                            printer.handleProgress(runningTotal, inputSize);
+                        }
+                    } catch (IOException e) {
+                        System.err.println("Could not download file. IO exception:" + e.getMessage());
+                        return false;
+                    }
+
+                }
+            }
+
+            writeTo.close();
```

Comment:
Would be cleaner with a try catch with resources

Reply:
added try catch with resources